docs/Overview.md

```diff
@@ -0,0 +1,1735 @@
+        }
+    }
+}
+```
+
+The messages are then transported over TCP. In .NET the standard TCP client and TCP server is used. A single *dynamic* port is used for a single pair of client and server. (Unlike in a more typical web scenario where server has a determined port, and a client connects to that port.)
+
+The role of client and server are not fully set, and based on parameters either the runner, or the testhost can serve the role of a server. Typically the runner is the server (setups a dynamic port, and starts testhost, passing the port to connect to). But in remote scenarios it is beneficial to make the remote side (testhost), the server, and open the port on the remote device which is more likely to be for-testing-only, and deprovisioned after the test run. In this case the server (testhost) needs to use a predetermined port which it sets up, and the client (runner) will connect to it. 
+
+
+### Header Part
+
+There is no header in the message itself. There is header only in the binary message.
+
+### Content Part
+
+
+
+### Base Types
+
+### Request Message
+
+### Response Message
+
+### Notification Message
+
+### Cancellation Support
+
+### Progress Support
+
+## Test Platform Protocol
+
+TestPlatform protocol defines a set of JSON request, response and notification messages, that are exchanged using the above base protocol. This section starts by describing the basic JSON structures used in the protocol. The description uses C# classes, and types, with nullability enabled. Meaning that every type is non-nullable by default, and nullability is denoted by `?` following the type name.
+
+<TODO example?>
+
+The protocol assumes that one server serves one tool. There is no support in the protocol to share one server between different tools.
+
+### Capabilities
+
+The client, runner + datacollector, and testhost are shipped separately. Each of those components can have a different version and hence a different set of functionality they support. A single number (protocol version) is used to represent the whole set of capabilities that a given component supports. Each newer version includes complete functionality of the previous version. There are no granular capabilities.
+
+Components are expected to handshake a version of the protocol to use, and use the highest version that is common to both sides.
+
+> For example client announces that is supports version 7, runner supports 9, version 7 is used.
+
+Once a version is agreed upon, a higher version should not be used by the lower level components. 
+
+> The above runner should announce 7 to testhost, rather than 9. If testhost announces 10, 7 is used. 
+
+The version is negotiated between the components at the beginning of every workflow using [ProtocolVersion](#protocolversion-request) request.
+
+### Request, Notification and Response Ordering
+
+The server supports processing only a single request at a time. Unless the request is [Cancel](#cancel) or [Abort](#abort) request.
```

Comment:
`Cancel` and `Abort` links are broken.

Reply:
Yes, how we handle cancellation and abort is missing from the document. Keeping as is for now.